README.md

```diff
@@ -11,18 +11,44 @@ Before using this module, make sure you have the following:
 
 ## Usage
 
-To use this module in your Terraform configuration, add the following code:
-
+To use this module in your Terraform configuration, add the following code in your existing terraform code:
 ```hcl
+module "scanner_role" {
+  source = "git::https://github.com/DataDog/terraform-module-datadog-agentless-scanner//modules/agentless-scanner-role"
+
+  account_roles       = [module.delegate_role.role.arn]
+  api_key_secret_arns = [module.agentless_scanner.api_key_secret_arn]
+}
+
+module "delegate_role" {
+  source = "git::https://github.com/DataDog/terraform-module-datadog-agentless-scanner//modules/scanning-delegate-role"
+
+  scanner_roles = [module.scanner_role.role.arn]
+}
+
 module "agentless_scanner" {
-  source = "github.com/DataDog/terraform-module-datadog-agentless-scanner"
+  source = "git::https://github.com/DataDog/terraform-module-datadog-agentless-scanner"
 
-  api_key               = "YOUR_API_KEY"
-  instance_profile_name = ""
+  api_key               = "YOUR API KEY"
+  instance_profile_name = module.scanner_role.instance_profile.name
 }
 ```
 
-## Example
+And run:
+```
+```
+
+> [!IMPORTANT]
+> We strongly recommend [pinning](https://developer.hashicorp.com/terraform/language/modules/sources#selecting-a-revision) the version of the module to keep repeatable deployment and to avoid unexpected changes.
```

Comment:
```suggestion
> Datadog strongly recommends [pinning](https://developer.hashicorp.com/terraform/language/modules/sources#selecting-a-revision) the version of the module to keep repeatable deployment and to avoid unexpected changes.
```